web/src/search/input/toggles/Toggles.tsx

```diff
@@ -0,0 +1,155 @@
+import * as React from 'react'
+import * as H from 'history'
+import RegexIcon from 'mdi-react/RegexIcon'
+import FormatLetterCaseIcon from 'mdi-react/FormatLetterCaseIcon'
+import { PatternTypeProps, CaseSensitivityProps } from '../..'
+import { FiltersToTypeAndValue } from '../../../../../shared/src/search/interactive/util'
+import { SettingsCascadeProps } from '../../../../../shared/src/settings/settings'
+import { SearchPatternType } from '../../../../../shared/src/graphql/schema'
+import { isEmpty } from 'lodash'
+import { submitSearch } from '../../helpers'
+import { Toggle } from './Toggle'
+import { isErrorLike } from '../../../../../shared/src/util/errors'
+import CodeBracketsIcon from 'mdi-react/CodeBracketsIcon'
+import { ActivationProps } from '../../../../../shared/src/components/activation/Activation'
+
+export interface TogglesProps extends PatternTypeProps, CaseSensitivityProps, SettingsCascadeProps {
+    navbarSearchQuery: string
+    history: H.History
+    location: H.Location
+    filtersInQuery?: FiltersToTypeAndValue
+    hasGlobalQueryBehavior?: boolean
+}
+
+interface SubmitSearchArgs {
+    history: H.History
+    navbarQuery: string
+    source: 'home' | 'nav' | 'repo' | 'tree' | 'filter' | 'type'
+    patternType: SearchPatternType
+    caseSensitive: boolean
+    activation?: ActivationProps['activation']
+    filtersQuery?: FiltersToTypeAndValue
+}
+
+export const Toggles: React.FunctionComponent<TogglesProps> = (props: TogglesProps) => {
+    const submitOnToggle = (args: SubmitSearchArgs): void => {
+        const searchQueryNotEmpty =
+            props.navbarSearchQuery !== '' || (props.filtersInQuery && !isEmpty(props.filtersInQuery))
+        const shouldSubmitSearch = props.hasGlobalQueryBehavior && searchQueryNotEmpty
+
+        if (shouldSubmitSearch) {
+            // Only submit search on toggle when the query input has global behavior (i.e. it's on the main search page
+            // or global navbar). Non-global inputs don't have the canonical query and need more context, making
+            // submit on-toggle undesirable. Also, only submit on toggle only when the query is non-empty.
+            submitSearch(
+                args.history,
+                args.navbarQuery,
+                args.source,
+                args.patternType,
+                args.caseSensitive,
+                args.activation,
+                args.filtersQuery
+            )
+        }
+    }
+
+    const ToggleCaseSensitivity = (): void => {
```

Comment:
Function names should be camelCase, not PascalCase.

```suggestion
    const toggleCaseSensitivity = (): void => {
```